plugin/src/functionalTest/kotlin/com/ryandens/javaagent/JavaagentPluginFunctionalTest.kt

```diff
@@ -95,6 +99,39 @@ DEFAULT_JVM_OPTS="-javaagent:${"$"}APP_HOME/lib/simple-agent.jar -Xmx256m"
         assertTrue(result.output.contains("Hello from my simple agent!"))
     }
 
+    @Test fun `can create docker image using jib`() {
+        val dependencies = """
+            javaagent project(':simple-agent')
+            runtimeOnly 'commons-lang:commons-lang:2.6'
+        """
+
+        // create the test project and run the tasks
+        val result = createAndBuildJavaagentProject(dependencies, listOf("jibBuildTar"))
+
+        // Verify the result
+        assertTrue(result.output.contains("BUILD SUCCESSFUL"))
+
+        // verify the agent was added after layer 1 in the Docker image
+        assertTrue(File(functionalTestDir, "hello-world/build/jib-image.tar").exists())
+        FileInputStream(File(functionalTestDir, "hello-world/build/jib-image.tar")).use { fis ->
+            ArchiveStreamFactory().createArchiveInputStream("tar", fis).use { ais ->
+                ais.nextEntry as TarArchiveEntry? // first layer (base image)
+                ais.nextEntry as TarArchiveEntry? // second layer (should be agent)
+                var agentFound = false
+                ArchiveStreamFactory().createArchiveInputStream("tar", GzipCompressorInputStream(ais)).use { l1ais ->
+                    var l1file = l1ais.nextEntry as TarArchiveEntry?
+                    while (l1file != null) {
+                        if (l1file.name.endsWith("HelloWorld.class")) {
+                            agentFound = true
+                        }
+                        l1file = l1ais.nextEntry as TarArchiveEntry?
+                    }
+                }
+                assertTrue(agentFound)
```

Comment:
Hmm, I'm wondering if this test is testing what it looks like it's doing. Based on how it's written, I would expect it to fail, because the layer should only be added if the `javaagent-jib` plugin is applied to the project, which it doesn't seem to be.   I generally very much agree that the functional test should test the docker component of this plugin, but checking for the individual layer position feels like something we can unit test and treating the functional tests a bit more like a fire alarm that only fails if the jib extension stops working for a valid use case. If you don't have a time to write a test like this i totally understand, and I can pick up where you left off with that. But, as it stands, this test i think gives me false confidence that the javaagent-jib plugin is working as intended. If you think I misunderstood something here, let me know! I'll try out a snapshot build of the plugin on a sample project just to get a feel for the actual plugin changes, but I think I'm totally on board with those as they stand

Reply:
@ryandens  Hey, you're absolutely right, thank you for flagging this. I had not realised the jib plugin had a different id 🤦 
I've added two more tests, but I had to change the scope of the extension-api and jib-gradle-plugin from 'runtimeOnly' to 'implementation'. Without that change, the tests fail (and also my test-project is not able to apply the extension). Please double-check whether this is correct.

Reply:
Thanks! Yeah, we should try to avoid making that change. We can use some gradle trickery to append the jib plugins to the Gradle test kit classpath

Reply:
Thanks for the Gradle sorcery, I've applied your changes, they seem to work